src/EncompassRest.Tests/LoanRateLockTests.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using EncompassRest.Loans;
+using EncompassRest.Loans.RateLocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace EncompassRest.Tests
+{
+    [TestClass]
+    public class LoanRateLockTests : TestBaseClass
+    {
+        [TestMethod]
+        [ApiTest]
+        public async Task LoanSubmitLockRequest()
+        {
+            var client = await GetTestClientAsync();
+            var loan = new Loan(client);
+            var loanId = await client.Loans.CreateLoanAsync(loan);
+            try
+            {
+                await Task.Delay(1000);
+                var rateLockApi = loan.LoanApis.RateLocks;
+                var rateLocks = await rateLockApi.GetRateLocksAsync();
+                Assert.IsNotNull(rateLocks);
+                Assert.AreEqual(0, rateLocks.Count);
+                var rateLockRequest = new RateLockRequest()
+                {
+                    LockRequest = new LockRequest()
+                    {
+                        LockDate = DateTime.Today,
+                        LockNumberOfDays = 15
+                    }
+                };
+                await rateLockApi.SubmitRateLockRequestAsync(rateLockRequest, true);
+                var requestId = rateLockRequest.Id;
+                Assert.IsFalse(string.IsNullOrEmpty(requestId));
+                Assert.AreEqual(LockStatus.Requested, rateLockRequest.LockStatus);
+                await Task.Delay(1000);
+                rateLocks = await rateLockApi.GetRateLocksAsync();
+                Assert.IsNotNull(rateLocks);
+                Assert.AreEqual(1, rateLocks.Count);
+                Assert.AreEqual(rateLockRequest.LockRequest.LockNumberOfDays, rateLocks[0].LockRequest.LockNumberOfDays);
+                Assert.AreEqual(rateLockRequest.LockRequest.LockExpirationDate, rateLocks[0].LockRequest.LockExpirationDate);
+                Assert.AreEqual(rateLockRequest.RequestedBy.EntityId, rateLocks[0].RequestedBy.EntityId);
+                
```

Comment:
Please check for ExtensionData with the `AssertNoExtensionData` method.

Reply:
Done in 7dcc95c